core/roadmap.rst

```diff
@@ -0,0 +1,39 @@
+Roadmap
+-------
+
+I don't really think a strict roadmap is the way forward. They are quite
+cumbersome and suck the joy out of open-source. However, I do believe in having
+a broad and overarching plan detailing how things could play out.
+
+Obviously, these are quite volatile. They are subject to change frequently and
+you shouldn't rely on this at all. A better and more complete list would be the
+`coala issue tracker <https://github.com/coala/coala/issues>`_.
+
+Major Future Features
+=====================
+
+Aspects
+*******
+
+Aspects are a completely new way to code analysis. You can read more about this
+in `cEP #5 <https://github.com/coala/cEPs/blob/master/cEP-0005.md>`_.
+
+Various Performance Enhancements
+********************************
+
+Despite various improvements in the past, coala has a lot of room for
+improvement when it comes to speed. We spend a lot of CPU time in unnecessary
+operations and optimizing these can give huge margins.
+
+Next-gen Design
+***************
+
+This is an ongoing work. You can read more about it in
+`cEP #2 <https://github.com/coala/cEPs/blob/master/cEP-0002.md>`_.
+
+Supporting Tools' Improvements
+******************************
+
+coala-quickstart and coala-html are fairly advanced (although there's still a
+lot of work to be done). But they are no where near the desired usage; so this
+project aims to investigate and develop ways we can improve on that area.
```

Comment:
This is for all the points:
I'd love more specific points although at this point there is probably not a sufficient amount of research to do that :)
Being more specific helps I feel.